### DISCUSSION TITLE
Props naming convention

### DISCUSSION THREAD

**Discussion by Webbrother** [dotansimha/graphql-code-generator#8363] [Q&A] [answered]
Hello!
My graphQl endpoint provides an underscore naming, and I use `@graphql-codegen/cli` to generate `graphql.schema.json`.
My `graphql.schema.json` uses underscore naming of props:

Example:
```graphql
query forums {
  forum {
    created_at // <- This
    id
    title
  }
}

```

But I would like to have a `camelCase` naming of all props.

Example:
```graphql
query forums {
  forum {
    createdAt // <- This
    id
    title
  }
}

```

Is it possible to do something to change props naming? I was playing with `namingConvention` configuration but it doesn't affect to properties names. 

My current `codegen.ts`:

```typescript
const schemaUrl = process.env.NEXT_PUBLIC_GRAPHQL_HOST;

const config: CodegenConfig = {
  overwrite: true,
  schema: schemaUrl,
  documents: ['./api/**/*.graphql'],
  config: {
    avoidOptionals: {
      field: true,
    },
    namingConvention: { // I tried everything here, but it doesn't help
      transformUnderscore: true,
    },
  },
  generates: {
    './api/_graphql/graphql.tsx': {
      plugins: [
        {
          typescript: {
            namingConvention: {
              transformUnderscore: true,
            },
          },
        },
        'typescript-operations',
        'typescript-react-apollo',
      ],
    },
    './graphql.schema.json': {
      plugins: ['introspection'],
    },
  },
};
```

**Comment by ardatan** [comment, marked as answer]
Those naming conventions are done on the generated types. Codegen doesn't transform your schema;
You have to use something like GraphQL Mesh + Naming Convention transform;
https://www.the-guild.dev/graphql/mesh/docs/transforms/naming-convention
Or GraphQL Tools transforms;
https://www.graphql-tools.com/docs/schema-wrapping#renaming
